ovos_utils/events.py

```diff
@@ -1,9 +1,81 @@
-from ovos_utils.log import LOG
-from ovos_utils.messagebus import Message, FakeBus
 import time
 from datetime import datetime, timedelta
 from inspect import signature
 
+from ovos_utils.intents.intent_service_interface import to_alnum
+from ovos_utils.log import LOG
+from ovos_utils.messagebus import Message, FakeBus
+
+
+def unmunge_message(message, skill_id):
+    """Restore message keywords by removing the Letterified skill ID.
+    Args:
+        message (Message): Intent result message
+        skill_id (str): skill identifier
+    Returns:
+        Message without clear keywords
+    """
+    if isinstance(message, Message) and isinstance(message.data, dict):
+        skill_id = to_alnum(skill_id)
+        for key in list(message.data.keys()):
+            if key.startswith(skill_id):
+                # replace the munged key with the real one
+                new_key = key[len(skill_id):]
+                message.data[new_key] = message.data.pop(key)
+
+    return message
+
+
+def get_handler_name(handler):
+    """Name (including class if available) of handler function.
+
+    Args:
+        handler (function): Function to be named
+
+    Returns:
+        string: handler name as string
+    """
+    if '__self__' in dir(handler) and 'name' in dir(handler.__self__):
+        return handler.__self__.name + '.' + handler.__name__
+    else:
+        return handler.__name__
+
+
+def create_wrapper(handler, skill_id, on_start, on_end, on_error):
```

Comment:
It might be helpful for future devs if the docstring explicitly states that `on_start` and `on_end` *must* accept the message as a positional arg